Introduce installation guide

PascalPrecht opened the pull request:
I'm working my way through the docs to learn more about what Embark can do and how it can be used. As I go along I make some suggestions for documentation. 

None of this is a must, so please feel free to take what you need, or nothing if that's what feels best to you 🙂

Please see inline comments for reasoning below.

PascalPrecht commented on source/docs/index.md:
```diff
@@ -1,102 +1,23 @@
+* **Client development** - Smart Contracts are not a must! Build your application with the framework of your choice right within Embark.
+* **Testing** - Test your applications and Smart Contracts through Web3 in JavaScript
+* **Decentralized app distribution** - Embark integrates with decentralized storages like IPFS and helps you distributing your app in the network.
+* **Peer-to-peer messaging** - Send and receive messages via communication protocols like Whisper and Orbit
```
Notice that I've reduced the list of features to a few more general ones, just to tease the reader with what Embark can do. I think not everyone will read through every single bit in this list anyways, so it's prolly better to keep it short and let the user explore if there there's interested in more in-depth topics (obviously, it's our job to have dedicated guides for everything)

jrainville replied:
Like above, should remove Orbit for now

PascalPrecht commented on source/docs/index.md:
```diff
@@ -1,102 +1,23 @@
-* Deploy the full application to IPFS or Swarm.
+There's so much more, but let's take it one step at a time. Make sure to check out our other guides in case you're looking for a particular feature.
 
+If you haven't yet, this is a good time to install Embark on your machine. Read on for more information!
```
It might make sense to introduce another "Useful Resources" section here with links to gitter, github etc, just so we have quick access for the reader without her parsing the whole text above for links.

alaibe replied:
Shall this be a link to how to install embark here?

PascalPrecht commented on source/docs/installation.md:
```diff
@@ -0,0 +1,66 @@
+
+In case you want to run a real node, [geth](https://geth.ethereum.org/) is a pretty good one. Check out the [installation guide](https://ethereum.github.io/go-ethereum/install/) for your platform and verify your installation with:
+
+<pre><code class="shell">$ geth --version</code></pre>
```
I thought it's better to just refer to the existing installation instructions from geth so we don't have to maintain those. Let me know if this should be changed.

andremedeiros replied:
Can we include the output from the tool? I want to see what to look for when I run this command.

PascalPrecht replied:
Absolutely. Let's add this.

PascalPrecht commented on source/docs/installation.md:
```diff
@@ -0,0 +1,66 @@
+
+[Ganache CLI](https://truffleframework.com/ganache) is probably the best client for development purposes as it emulates a blockchain and is therefore fast than a real node:
+
+<pre><code class="shell">$ npm -g install ganache-cli</code></pre>
```
I put ganache-cli first here, as it's probably the fastest option for pure development and getting stuff going as quickly as possible. It also shows how we greatly integrate with other tools!

alaibe replied:
I believe you don't even need that, because you can just run embark simulator and it will start the simulator using ganache

PascalPrecht replied:
Thank you!

Yes I can confirm that. Just checked the source: https://embark.status.im/docs/embark_commands.html#simulator

That's actually pretty cool as it's less setup for the user! :)

PascalPrecht commented on source/docs/installation.md:
```diff
@@ -0,0 +1,66 @@
+To verify that the installation was successful, simply run the following command:
+
+<pre><code class="shell">$ ipfs --version
+</code></pre>
```
We had a lot of additional information about IPFS and how its headers need to be configured etc.

I don't think this is needed anymore, so I've removed that part.

andremedeiros replied:
Can we include the output from the tool? I want to see what to look for when I run this command.

alaibe commented on source/docs/installation.md:
```diff
@@ -0,0 +1,66 @@
+
+
+
+
```
I don't think you need all those extra empty lines?

PascalPrecht commented:
Thanks for the review @alaibe 

Addressed your comments and updated the PR

andremedeiros commented on source/docs/installation.md:
```diff
@@ -0,0 +1,59 @@
+
+## Prerequisites
+
+In order to make Embark work on our machine, we need to have some tools installed first. Make sure you have the following ready and in the correct version:
```
`on our machine` -> `in your computer`

PascalPrecht replied:
This sounds better, thanks!

andremedeiros commented on source/docs/installation.md:
```diff
@@ -0,0 +1,59 @@
+
+After that, `embark` should be available as a global command in your terminal of choice. Let's verify this by running the following command:
+
+<pre><code class="shell">$ embark --version</code></pre>
```
Can we include the output from the tool? I want to see what to look for when I run this command.

jrainville replied:
It's actually `embark version`

PascalPrecht replied:
uhm.. `embark --version` does the trick though

PascalPrecht replied:
Both do the same, but if you prefer going with `version`, I can update this.

jrainville replied:
Really? Was this added recently? Anyway, TIL
You can keep it as is then.

PascalPrecht commented:
Thanks for your comments @andremedeiros ! I've adjusted the content accordingly and updated the PR.

Feel free to give this another shot :)

PascalPrecht commented:
Looping in @iurimatias to give it green light before merge

PascalPrecht commented on source/docs/dashboard.md:
```diff
@@ -0,0 +1,20 @@
+if you prefer to only see the logs, you can disable the dashboard with the nodashboard option ``embark run --nodashboard``
+{% endnote %}
+
+Note that if you update your code, it will automatically be re-deployed, contracts included. There is no need to restart embark, refreshing the page on the browser will do.
```
As mentioned in the commit message, this is just extracted from the quickstart for now but will be polished in follow up commits.

In other words, we don't lose this content, we just moved it somewhere else so we can keep the quickstart as focused as possible.

PascalPrecht commented on source/_data/sidebar.yml:
```diff
@@ -1,11 +1,14 @@
   general_usage:
     creating_project: create_project.html
     structure: structure.html
+    running_apps: running_apps.html
```
This is a new guide that focuses on things like options for `embark run` as well as environments and the simulator.

PascalPrecht commented on source/docs/quick_start.md:
```diff
@@ -1,61 +1,38 @@
 
 <pre><code class="shell">$ embark demo
-$ cd embark_demo
+$ cd embark demo
```
This needs to be fixed.

PascalPrecht commented:
@andre @iurimatias I've added another commit that updates the quickstart + it introduces a new "Running apps" guide as well.

PTAL

PascalPrecht commented on source/docs/quick_start.md:
```diff
@@ -1,61 +1,38 @@
-**available services**
-
-Available Services will display the services available to your dapp in green. If a service is down, then it will be displayed in red.
+For now, let's focus on what has happened in the meantime. Embark has compiled and deployed the Smart Contracts that come with the demo application to a custom blockchain on your local machine. It has also compiled the web app that's part of the demo and deployed that to a local web server, which is listening on `http://localhost:8000`. In fact, Embark has probably already opened a browser window ...
```
The dashboard screenshot in line 26 probably should be updated with a more "neutral" version. For example, the first time I saw the embark docs it looked like embark comes with a mountain landscape out of the box. 

Some people may not like this idea and might refuse to look further into the tool ( I know sounds silly, but let's just be sure)

Also, we should ensure that the font size in the screenshot is big enough so whatever is being shown is easy to read.

PascalPrecht replied:
This is done in https://github.com/embark-framework/embark-site/pull/82/commits/217b4554bb2e00822738d7375c9630b8572c6f70

PascalPrecht commented on source/docs/running_apps.md:
```diff
@@ -0,0 +1,96 @@
+
+Another feature of Embark is to start a **simulated** blockchain. This can be useful for testing purposes as there's no need to wait for transactions to be mined. You might have heard of [Ganache CLI](https://truffleframework.com/docs/ganache/quickstart), which is a great project that implements such a simulated blockchain.
+
+Embark integrates perfectly with this existing tool chain. To start a simulated blockchain, all we have do is to use the `simulator` command:
```
I personally like getting the message across that we're open to collaborate and integrate with other tools. Let me know what you think about this wording.

andremedeiros commented on source/docs/dashboard.md:
```diff
@@ -0,0 +1,20 @@
+title: Working with the dashboard
+---
+
+![Dashboard](http://i.imgur.com/s4OQZpu.jpg)
```
This image should live in the repo.

PascalPrecht replied:
Agreed. Also as mentioned in https://github.com/embark-framework/embark-site/pull/82#discussion_r221361496 I think we should swap it out with a more "neutral" version of it.

andremedeiros commented on source/docs/dashboard.md:
```diff
@@ -0,0 +1,20 @@
+
+The dashboard will tell you the state of your contracts, the environment you are using, and what Embark is doing at the moment.
+
+**available services**
```
Capitalization

andremedeiros commented on source/docs/dashboard.md:
```diff
@@ -0,0 +1,20 @@
+
+Available Services will display the services available to your dapp in green. If a service is down, then it will be displayed in red.
+
+**logs and console**
```
Capitalization

andremedeiros commented on source/docs/index.md:
```diff
@@ -1,102 +1,22 @@
 ## What is Embark?
 
-Embark is a fast, simple and powerful framework to develop and deploy Decentralized Applications (DApps). 
+Embark is a fast, easy to use and powerful developer environment to build and deploy decentralized applications, also known as "DApps". It integrates with [Ethereum](https://ethereum.org/) blockchains, decentralized storages like [IPFS](https://ipfs.io) and [Swarm](https://swarm-guide.readthedocs.io/en/latest/introduction.html), and decentralized communication platforms like [Whisper](https://git ...
```
I feel that this should either read "easy to use but powerful" or "easy to use, and powerful". Not 100% sure on that tho.

PascalPrecht replied:
Updated to `easy to use, and powerful`

andremedeiros commented on source/docs/index.md:
```diff
@@ -1,102 +1,22 @@
+* **Automatic Smart Contract deployment** - Embark takes care of deploying your Smart Contracts as well as redeploying them as you make changes to your code.
+* **Client development** - Smart Contracts are not a must! Build your application with the framework of your choice right within Embark.
+* **Testing** - Test your applications and Smart Contracts through Web3 in JavaScript
+* **Decentralized app distribution** - Embark integrates with decentralized storages like IPFS and helps you distributing your app in the network.
```
`like IPFS and Swarm` perhaps?

PascalPrecht replied:
This has landed in the PR as well!

andremedeiros commented on source/docs/quick_start.md:
```diff
@@ -1,61 +1,38 @@
 </code></pre>
 
-### Start a blockchain node or let Embark start it for you
+This will create a demo application. Feel free to look around to get familiar with the project's structure, but don't worry, we'll take a closer look at it later in one of our [dedicated guides](structure.html).
```
`but don't worry, we'll take a closer look` -> `but don't worry. We'll take a closer look`

PascalPrecht replied:
Updated!

PascalPrecht commented:
Thanks for the review @andremedeiros ! I'll address the comments in the dashboard doc in a follow up commit that focuses entirely on that part of the docs.

PascalPrecht commented:
Added another commit that updates the documentation on creating apps. Let me know what you think

PascalPrecht commented:
@andremedeiros I've also put some love into the dashboard guide now (with an updated screenshot). 

PTAL

michaelsbradleyjr commented on source/docs/installation.md:
```diff
@@ -0,0 +1,79 @@
+
+Once that is done, you can install and select a specific Node version using the `nvm` command like this:
+
+<pre><code class="shell">$ nvm install 8.11
```
maybe we should swap `8.11` for `--lts` — valid for `nvm install` and `nvm use` commands.

PascalPrecht replied:
I had no idea this option existed. Thanks, that's much better indeed :)

jrainville commented on source/docs/index.md:
```diff
@@ -1,102 +1,22 @@
 ## What is Embark?
 
-Embark is a fast, simple and powerful framework to develop and deploy Decentralized Applications (DApps). 
+Embark is a fast, easy to use, and powerful developer environment to build and deploy decentralized applications, also known as "DApps". It integrates with [Ethereum](https://ethereum.org/) blockchains, decentralized storages like [IPFS](https://ipfs.io) and [Swarm](https://swarm-guide.readthedocs.io/en/latest/introduction.html), and decentralized communication platforms like [Whisper](https://gi ...
```
Should probably remove Orbit for now, as we don't support it right now

PascalPrecht replied:
Done :)

jrainville commented on source/docs/index.md:
```diff
@@ -1,102 +1,22 @@
-* Manage different chains (e.g testnet, private net, livenet)
-* Easily manage complex systems of interdependent contracts.
+* **Automatic Smart Contract deployment** - Embark takes care of deploying your Smart Contracts as well as redeploying them as you make changes to your code.
+* **Client development** - Smart Contracts are not a must! Build your application with the framework of your choice right within Embark.
```
Not sure what you mean by that? Smart contracts are pretty important, so not sure if you meant Solidity is not a must

PascalPrecht replied:
Oh sorry, yea this probably needs to be reworded. And also, I think I had a brain twist here.

The point was that you can use Embark and decentralization features without actually making your app dependent on Smart Contracts. E.g. if you just want to build a SPA that's deployed via IPFS, you could do that. 

But maybe we want to turn this around and focus on the fact that you can just deploy Smart Contracts without building a SPA?

jrainville replied:
Ah ok, I understand. Maybe just leave the smart contract part out and just talk about Front end development using the framework of your choice.

jrainville commented on source/docs/installation.md:
```diff
@@ -0,0 +1,79 @@
+
+- [Node](#Node)
+- [IPFS](#IPFS-Optional)
+- [Ethereum Client](#Ethereum-Client-Optional)
```
I would put Geth above IPFS

PascalPrecht replied:
Done!

jrainville commented on source/docs/installation.md:
```diff
@@ -0,0 +1,79 @@
+
+## Installing Embark
+
+Alright, let's install Embark so we can build our first application. As mentioned earlier, if anything is unclear or you run into problems, make sure to reach out to us on our dedicated channels or [submit an issue on  GitHub](https://github.com/embark-framework/embark/issues).
```
You could link to Troubleshooting too

PascalPrecht replied:
Good point!

jrainville commented on source/docs/installation.md:
```diff
@@ -0,0 +1,79 @@
+
+At the time of writing this guide, the output looked like this:
+
+<pre><code class="shell">3.2.1</code></pre>
```
You should put `3.2.x`

PascalPrecht commented on source/docs/create_project.md:
```diff
@@ -1,57 +1,46 @@
 ---
 
-Once Embark is installed, you can create a new Embark project, you have many options to do so:
+Once Embark is installed, there are different ways to create a new decentralized application. Whether we intent to create an application that uses only Smart Contracts, or we want to take advantage of other decentralized features like storages and messaging. Embark provides options and templates for various scenarios. Let's take a look!
```
Good catch!

PascalPrecht commented on source/docs/create_project.md:
```diff
@@ -1,57 +1,46 @@
+## Using the `demo` command
 
-If you are first starting & exploring Embark you can create a demo project:
+As discussed in our [quickstart guide](quick-start.html), the fastest way to get up and running with Embark is using its `demo` command. This will scaffold a new demo application and its needed environment in a folder called `embark_demo`. The demo application lets play with Embark's APIs through a pre-built web interface.
```
I think it's "le**ts** you play" no?

PascalPrecht commented on source/docs/create_project.md:
```diff
@@ -1,57 +1,46 @@
+## Creating a new app
 
-To create a brand new project you can use:
+If you prefer starting entirely from scratch but still having a ready to use environment, Embark's `new` command has got you covered. Similar to the `demo` command, it will scaffold a new project folder. However, it will not include the demo application. The green field is all yours.
```
Reworded this!

PascalPrecht commented on source/docs/create_project.md:
```diff
@@ -1,57 +1,46 @@
-<pre><code class="shell">$ embark new &lt;YourDAppName&gt; --template angular
-$ cd &lt;YourDAppName&gt;
-</code></pre>
+In fact, in case of GitHub, the sam can be done with the username/repository shortcut:
```
👍

PascalPrecht commented on source/docs/dashboard.md:
```diff
@@ -0,0 +1,30 @@
+The dashboard is separated into different sections, each with their own responsibility. The sections are:
+
+- **Contracts** - This section shows you not only all the Smart Contracts within your project, but also their status on whether they are being deployed at the moment or not. Notice that, as we make chances to our Smart Contracts, the dashboard will reflect the automatic redeployment in this section as well, making it very easy to stay on top of things.
+  <br>
```
Yes and no. What I want to achieve as a newline after the list point, but with vanilla markdown, this is not possible. If we introduce two empty lines, this will actually result in one `<ul>` per `<li>` which is semantically incorrect. That's why I went with `<br>` for now.

Obviously, this should actually be tackled using CSS

PascalPrecht replied:
Fixed this with CSS now

PascalPrecht commented:
Thanks for the review @jrainville and @michaelsbradleyjr !

Addressed all of your comments and updated the commits accordingly.

@jrainville there are 2-3 points I still need your feedback on. Please take a look at the comments I responded to :)
